llvm/lib/CodeGen/AtomicExpandPass.cpp

```diff
@@ -153,42 +153,6 @@ static unsigned getAtomicOpSize(AtomicCmpXchgInst *CASI) {
   return DL.getTypeStoreSize(CASI->getCompareOperand()->getType());
 }
 
-// Helper functions to retrieve whether an atomic instruction is operating on a
-// capability value.
-static unsigned getAtomicOpValIsCap(LoadInst *LI) {
-  const DataLayout &DL = LI->getModule()->getDataLayout();
-  return DL.isFatPointer(LI->getType());
-}
-
-static unsigned getAtomicOpValIsCap(StoreInst *SI) {
-  const DataLayout &DL = SI->getModule()->getDataLayout();
-  return DL.isFatPointer(SI->getValueOperand()->getType());
-}
-
-static unsigned getAtomicOpValIsCap(AtomicRMWInst *RMWI) {
-  const DataLayout &DL = RMWI->getModule()->getDataLayout();
-  return DL.isFatPointer(RMWI->getValOperand()->getType());
-}
-
-static unsigned getAtomicOpValIsCap(AtomicCmpXchgInst *CASI) {
-  const DataLayout &DL = CASI->getModule()->getDataLayout();
-  return DL.isFatPointer(CASI->getCompareOperand()->getType());
-}
-
-// Determine if a particular atomic operation has a supported size,
-// and is of appropriate alignment, to be passed through for target
-// lowering. (Versus turning into a __atomic libcall)
-template <typename Inst>
-static bool atomicSizeSupported(const TargetLowering *TLI, Inst *I) {
```

Comment:
It'd be nice to push some of this refactoring upstream at least in terms of structure (I see no reason not to even upstream the full refactor minus the caps checks with the argument that we need them downstream and others might too, but they might prefer you drop unused arguments which is fine, we can still introduce them locally, it's a smaller diff).

Reply:
I'll start working on that now.